worker/mutation.go

```diff
@@ -195,6 +200,19 @@ func runSchemaMutation(ctx context.Context, updates []*pb.SchemaUpdate, startTs
 	var wg sync.WaitGroup
 	wg.Add(1)
 	defer wg.Done()
+	// This throttle allows is used to limit the number of files which are opened simultaneously
+	// by badger while building indexes for predicates in background.
+	maxOpenFileLimit, err := QueryMaxOpenFiles()
+	if err != nil {
+		// Setting to default value on unix systems
+		maxOpenFileLimit = 1024
+	}
+	glog.Infof("Max open files limit: %d", maxOpenFileLimit)
+	// Badger opens around 7 files for indexing per predicate.
```

Comment:
How did you calculate the number 7?

Reply:
This PR to address the error of "Too many open files" on empty database. I arrived at this number to account for the files, MANIFEST, LOCK, KEYREGISTRY, DISCARD, 00001.mem and 00001.vlog